Vagrant Script: Detect colliding active virtualbox VMs and warn users

vsk-coding opened the pull request:
By running vagrant script when there is active residual VMs that are not
visible to vagrant creates error.

This commit fix that error by listing the active residual VMs and
presenting the user with options to Pause, Stop or Destroy the residual
VMs or ignore and continue.

...
$NO_BUILD=1 K8S=1 ./contrib/vagrant/start.sh

Detected running VMs that might cause conflict

"cilium_k8s1_1610031382439_78313" {a3c160c3-9f2b-4aae-8c87-44e004a7a003}

Do you wish to pause, stop, or destroy the VM(s) or ignore and continue?
[p/s/d/C]
...

Fixes: #13928

Signed-off-by: Vishnu Soman K <vishnusomank05@gmail.com>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

<!-- Description of change -->

Fixes: #issue-number

pchaigno commented on contrib/vagrant/start.sh:
```diff
@@ -576,36 +576,110 @@ function vboxnet_addr_finder(){
+    VBoxManage list runningvms
+	#option to pause, stop or delete VM(s) or ignore and continue
+    echo
+	printf "Do you wish to pause, stop, or destroy the VM(s) or ignore and continue? [p/s/d/C]  "
```
This message is now printed twice (here and line 623 above).

pchaigno commented on contrib/vagrant/start.sh:
```diff
@@ -576,36 +576,110 @@ function vboxnet_addr_finder(){
+			printf "\n$runningVm VM(s) destroyed successfully\n"
+			printf "Tip: Try running vagrant status to check status\n\n"
+		;;
+		"C" )
```
We need to make this the default case.

pchaigno commented on contrib/vagrant/start.sh:
```diff
@@ -576,36 +576,110 @@ function vboxnet_addr_finder(){
+				echo $VMName
+				VBoxManage controlvm $VMName pause
+				if [ $(VBoxManage showvminfo $VMName | grep -c "paused (since") -eq 0 ]; then
+                    VBoxManage controlvm $VMName pause
```
Why are we trying to pause the VM twice?

pchaigno commented on contrib/vagrant/start.sh:
```diff
@@ -576,36 +576,110 @@ function vboxnet_addr_finder(){
+				fi
+			done
+			printf "\n$runningVm VM(s) paused successfully\n" 
+			printf "\n$temp VM(s) paused successfully\n" 
```
Looks like a duplicated message.

pchaigno commented on contrib/vagrant/start.sh:
```diff
@@ -576,36 +576,110 @@ function vboxnet_addr_finder(){
+	read optn
+    temp=$runningVm
+	case "$optn" in
+		"p" )
```
I still don't think `pause` is a viable option. If we choose to pause the VM on the prompt, then we still get the prompt again on the next `start.sh` run.

vishnusomank committed:
Vagrant Script: Detect colliding active virtualbox VMs and warn users

By running vagrant script when there is active residual VMs that are not
visible to vagrant creates error.

This commit fix that error by listing the active residual VMs and
presenting the user with options to Pause, Stop or Destroy the residual
VMs or ignore and continue.

...
$ NO_BUILD=1 K8S=1 NETNEXT=1 NWORKERS=1 ./contrib/vagrant/start.sh k8s2+
Detected running VMs that might cause conflict

"cilium_k8s1_1611423389090_29763" {f12ec3c4-4d88-46c4-a2e8-9b6a415436fa}
"cilium_k8s2_1611425788862_92022" {af0cbdac-991e-4f8d-82a0-9d12299a365e}
"cilium_k8s2_1611429983338_86249" {bd30955e-2260-4e9f-9289-5f974c986c18}
"cilium_k8s1_1611468585872_25900" {767f83c1-e1f4-489f-b911-db7a44418f23}

Do you wish to stop, destroy the VM(s) or ignore and continue? [s/d/C]
...

fixed styling issue and added newline

Fixes: #13928

Signed-off-by: Vishnu Soman K <vishnusomank05@gmail.com>

pchaigno commented on contrib/vagrant/start.sh:
```diff
@@ -576,36 +576,91 @@ function vboxnet_addr_finder(){
+    VBoxManage list runningvms
+
+	#option to stop, delete VM(s) or ignore and continue
+    echo
```
Could you fix up all indentations in this file? It looks like there's a mix of tabulations and spaces.

pchaigno commented on contrib/vagrant/start.sh:
```diff
@@ -576,36 +576,91 @@ function vboxnet_addr_finder(){
+    echo
+    #value to start.sh is passed to function createVm
+	createVm $1
+fi 
```
Missing newline here.

pchaigno reviewed (commented):
It looks great! Thanks @vsk-coding!

Only two minor code style issues to fix below. I think we'll be good after that.

christarazi reviewed (changes requested):
Thanks for the PR! Could you squash the two commits into one?
